special-pages/pages/new-tab/app/omnibar/integration-tests/omnibar.page.js

```diff
@@ -0,0 +1,148 @@
+import { expect } from '@playwright/test';
+
+export class OmnibarPage {
+    /**
+     * @param {import("../../../integration-tests/new-tab.page.js").NewtabPage} ntp
+     */
+    constructor(ntp) {
+        this.ntp = ntp;
+        this.page = this.ntp.page;
+    }
+
+    context() {
+        return this.page.locator('[data-entry-point="omnibar"]');
+    }
+
+    async ready() {
+        await this.ntp.mocks.waitForCallCount({ method: 'omnibar_getConfig', count: 1 });
+    }
+
+    searchInput() {
+        return this.context().getByRole('combobox');
+    }
+
+    chatInput() {
+        return this.context().getByRole('textbox', { name: /Chat privately with Duck\.ai/i });
+    }
+
+    tabList() {
+        return this.context().getByRole('tablist');
+    }
+
+    searchTab() {
+        return this.context().getByRole('tab', { name: /Search/i });
```

Comment:
These regexes are overly permissive, i'd match on exact strings until you have one with dynamic content. Even then we can focus on deterministic data/ui to avoid dismiss things like this where possible.

Reply:
caa3511173916b0301fa673a9e0ab59ff48e7638